lib/data.ml

```diff
@@ -24,18 +9,24 @@ module Eq = struct
   let ( = ) {M : Eq} = M.(=)
 end
 
+type ordering = LT | GT | EQ
+
 module type Ord = sig
   type t
-  val compare : t -> t -> int
+  val compare : t -> t -> ordering
 end
 
 module Ord = struct
-  let ( =  ) {M : Ord} a b = M.compare a b =  0
-  let ( <> ) {M : Ord} a b = M.compare a b <> 0
-  let ( <  ) {M : Ord} a b = M.compare a b <  0
-  let ( <= ) {M : Ord} a b = M.compare a b <= 0
-  let ( >  ) {M : Ord} a b = M.compare a b >  0
-  let ( >= ) {M : Ord} a b = M.compare a b >= 0
+  let translateCompare (n : int) =
+    if n < 0 then LT
+    else if n = 0 then EQ
+    else (* if n > 0 then *) GT
```

Comment:
From the name I expected this to be a translation function on comparisons (`(t -> t -> int) -> (t -> t -> ordering)`), and I think it'd be a bit more convenient to use that way